### PR TITLE
Feature/add server timing headers

### PR DESCRIPTION
**Pull request by supercid**

### CODE REVIEW

File: Plugin/Catalog/Block/Toolbar.php [hunk 1]
```diff
@@ -180,12 +181,19 @@ private function getCmpResult(Store $store, FulltextCollection $collection)
...
+        TimeHelper::getInstance()->instrument(
+            function () use (&$nostoAccount, &$nostoCustomer, &$categoryString, &$limit, &$personalizationResult) {
+                $personalizationResult = $this->categoryRecommendation->getPersonalisationResult(
+                    $nostoAccount,
+                    $nostoCustomer,
+                    $categoryString,
+                    $limit
+                );
+            },
+            'CMP'
```
Comment (mridang) [inline Plugin/Catalog/Block/Toolbar.php]: lowercase

File: Plugin/Catalog/Block/Toolbar.php [hunk 2]
```diff
@@ -180,12 +181,19 @@ private function getCmpResult(Store $store, FulltextCollection $collection)
...
         $nostoCustomer = $this->cookieManager->getCookie(NostoCustomer::COOKIE_NAME);
         $limit = $collection->getSize();
-        return $this->categoryRecommendation->getPersonalisationResult(
-            $nostoAccount,
-            $nostoCustomer,
-            $categoryString,
-            $limit
+        $personalizationResult = null;
+        TimeHelper::getInstance()->instrument(
+            function () use (&$nostoAccount, &$nostoCustomer, &$categoryString, &$limit, &$personalizationResult) {
```
Comment (hannunosto) [inline Plugin/Catalog/Block/Toolbar.php]: I don't think we need to use the `&`

File: Plugin/Catalog/Block/Toolbar.php (hunk 1 above)
Comment (hannunosto) [inline Plugin/Catalog/Block/Toolbar.php]: Perhaps use constant here

File: Observer/App/Action/Action.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        HttpResponse $response,
+        HttpRequest $request
+    )
+    {
+        $this->response = $response;
+        $this->request = $request;
+    }
+    public function execute(Observer $observer)
+    {
+        if ((string)$this->request->getFullActionName() === self::CATALOG_PAGE_TYPE) {
```
Comment (hannunosto) [inline Observer/App/Action/Action.php]: Do we need to check the page type? I would just check that if there are elements (we need a checker method in `TimeHelper`) in `TimeHelper` and if there are, add it to the header

File: Helper/TimeHelper.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        $fn();
+        $stop = microtime(true);
+        $this->times[$name] = $stop - $start;
+    }
+
+    public function build()
+    {
+        $value = '';
+        foreach ($this->times as $key => $time) {
+            $value .= $key . ';dur=' . $time . ',';
```
Comment (hannunosto) [inline Helper/TimeHelper.php]: `sprintf` might be easier to read

**Pull request merged**